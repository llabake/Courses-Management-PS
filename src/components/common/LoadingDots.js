import React, { Component } from 'react';
import PropTypes from 'prop-types';

class LoadingDots extends Component {
  state = {
    frame: 1,
  };

  componentDidMount() {
    this.interval = setInterval(() => {
      this.setState({ // eslint-disable-line react/no-did-mount-set-state
        // eslint-disable-next-line react/no-access-state-in-setstate
        frame: this.state.frame + 1,
      });
    }, this.props.interval);
  }

  componentWillUnmount() {
    clearInterval(this.interval);
  }

  render() {
    let dots = this.state.frame % (this.props.dots + 1);
    let text = '';

    while (dots > 0) {
      text += '.';
      // eslint-disable-next-line no-plusplus
      dots--;
    }

    return (
      <span {...this.props}>
        {text}
&nbsp;
      </span>
    );
  }
}

LoadingDots.propTypes = {
  interval: PropTypes.number,
  dots: PropTypes.number,
};

LoadingDots.defaultProps = {
  interval: 300,
  dots: 3,
};

export default LoadingDots;
