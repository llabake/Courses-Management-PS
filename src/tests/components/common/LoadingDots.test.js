import React from 'react';
import { shallow } from 'enzyme';
import LoadingDots from "../../../components/common/LoadingDots";

describe('<LoadingDots />', () => {
  it('renders correctly', () => {
    const props = {
      interval: 300,
      dots: 4
    };
    const wrapper = shallow(<LoadingDots {...props}/>);
    expect(wrapper).toMatchSnapshot();
  });
});
