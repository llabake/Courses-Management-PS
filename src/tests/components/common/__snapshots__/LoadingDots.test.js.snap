// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoadingDots /> renders correctly 1`] = `
<span
  dots={4}
  interval={300}
>
  .
  Â 
</span>
`;
